modules/navier_stokes/src/executioners/SIMPLE.C

```diff
@@ -0,0 +1,1279 @@
+  NonlinearImplicitSystem & se_system =
+      libMesh::cast_ref<NonlinearImplicitSystem &>(_solid_energy_system->system());
+
+  // We will need the solution, the right hand side and the matrix
+  NumericVector<Number> & current_local_solution = *(se_system.current_local_solution);
+  NumericVector<Number> & solution = *(se_system.solution);
+  SparseMatrix<Number> & mat = *(se_system.matrix);
+  NumericVector<Number> & rhs = *(se_system.rhs);
+
+  // Fetch the linear solver from the system
+  PetscLinearSolver<Real> & se_solver =
+      libMesh::cast_ref<PetscLinearSolver<Real> &>(*se_system.get_linear_solver());
+
+  // We need a zero vector to be able to emulate the Ax=b system by evaluating the
+  // residual and jacobian. Unfortunately, this will leave us with the -b on the righ hand side
+  // so we correct it by multiplying it with (-1)
+  auto zero_solution = current_local_solution.zero_clone();
+  _problem.computeResidualAndJacobian(*zero_solution, rhs, mat);
+  rhs.scale(-1.0);
+
+  if (_print_fields)
+  {
+    _console << "Solid energy matrix" << std::endl;
+    mat.print();
+  }
+
+  // We compute the normalization factors based on the fluxes
+  Real norm_factor = computeNormalizationFactor(solution, mat, rhs);
+
+  // We need the non-preconditioned norm to be consistent with the norm factor
+  KSPSetNormType(se_solver.ksp(), KSP_NORM_UNPRECONDITIONED);
+
+  // Setting the linear tolerances and maximum iteration counts
+  _solid_energy_linear_control.real_valued_data["abs_tol"] = _solid_energy_l_abs_tol * norm_factor;
+  se_solver.set_solver_configuration(_solid_energy_linear_control);
+
+  se_solver.solve(mat, mat, solution, rhs);
+  se_system.update();
+
+  if (_print_fields)
+  {
+    _console << " Solid energy rhs " << std::endl;
+    rhs.print();
+    _console << " Solid temperature " << std::endl;
+    solution.print();
+    _console << "Norm factor " << norm_factor << std::endl;
+  }
+
+  _solid_energy_system->setSolution(current_local_solution);
+
+  return se_solver.get_initial_residual() / norm_factor;
+}
+
+void
+SIMPLE::execute()
+{
+  if (_app.isRecovering())
+  {
+    _console << "\nCannot recover SIMPLE solves!\nExiting...\n" << std::endl;
+    return;
+  }
+
+  if (_problem.adaptivity().isOn())
+  {
+    _console << "\nCannot use SIMPLE solves with mesh adaptivity!\nExiting...\n" << std::endl;
+    return;
+  }
+
+  if (_problem.hasMultiApps(EXEC_TIMESTEP_BEGIN) || _problem.hasMultiApps(EXEC_TIMESTEP_END) ||
+      _problem.hasMultiApps(EXEC_INITIAL) ||
+      _problem.hasMultiApps(EXEC_MULTIAPP_FIXED_POINT_BEGIN) ||
+      _problem.hasMultiApps(EXEC_MULTIAPP_FIXED_POINT_END))
+  {
+    _console << "\nCannot use SIMPLE solves with MultiApps set to execute on TIMESTEP_BEGIN, "
+                "TIMESTEP_END, INITIAL, MULTIAPP_FIXED_POINT_BEGIN or "
+                "MULTIAPP_FIXED_POINT_END!\nExiting...\n"
+             << std::endl;
+    return;
+  }
+
```

Comment:
exec on LINEAR and NONLINEAR are ignored. eithr add to the docs about it or check all objects here (might be a pain)

Reply:
Adding to the docs